fetch.js

```diff
@@ -338,6 +338,7 @@ export function Request(input, options) {
   this.mode = options.mode || this.mode || null
   this.signal = options.signal || this.signal
   this.referrer = null
+  this.cf = options.cf || {}
```

Comment:
I think we need to be stricter about what is possible on the `cf` object otherwise the behaviour doesn't align with the Cloudflare implementation. My proposal here is that we drop any keys that are not defined in the [controlling Cloudflare feature docs](https://developers.cloudflare.com/workers/reference/cloudflare-features/) to ensure that if a key is set, it will remove it and *cannot* be asserted against. Without this, someone could any key to this object, test against it but not have it do anything within the worker environment.

Reply:
+1